# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/FinalizationObservers.h
# Commit: 8e7816b5ea2a
# Full Hash: 8e7816b5ea2a6e9abd80ef165141cb5f8fd6a2d8
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-01 16:17:59
# Regressor Bug: 1755725
# File Overlap Count: 1
# Description:
#   Bug 1755725 - Part 2: Refactor FinalizationObservers in preparation to add a weak set for weak refs r=sfink
#   
#   This renames crossZoneWarppers to crossZoneRecordWrappers and makes
#   add/RemoveCrossZoneWarpper take the set as an argument.
#   
# ==============================================================================

diff -r 59ae32a4a0a3 -r 8e7816b5ea2a js/src/gc/FinalizationObservers.h
--- a/js/src/gc/FinalizationObservers.h	Tue Mar 01 08:28:41 2022 +0000
+++ b/js/src/gc/FinalizationObservers.h	Tue Mar 01 08:28:41 2022 +0000
@@ -67,7 +67,7 @@
   // marking rules keep the wrappers alive while the record is alive and ensure
   // that they are both swept in the same sweep group.
   using WrapperWeakSet = ObjectValueWeakMap;
-  WrapperWeakSet crossZoneWrappers;
+  WrapperWeakSet crossZoneRecords;
 
   // A map of weak ref targets to a vector of weak refs that are observing the
   // target. The weak refs may be in other zones and are wrapped appropriately.
@@ -98,8 +98,8 @@
 #endif
 
  private:
-  bool addCrossZoneWrapper(JSObject* wrapper);
-  void removeCrossZoneWrapper(JSObject* wrapper);
+  bool addCrossZoneWrapper(WrapperWeakSet& weakSet, JSObject* wrapper);
+  void removeCrossZoneWrapper(WrapperWeakSet& weakSet, JSObject* wrapper);
 
   static bool shouldRemoveRecord(FinalizationRecordObject* record);
 };
